aten/src/ATen/native/nested/NestedTensorTransformerFunctions.cpp

```diff
@@ -246,5 +246,73 @@ Tensor NestedTensor_to_mask(const Tensor& nt, std::optional<int64_t> mask_dim, c
   return result;
 }
 
+Tensor _jagged_to_padded_dense_forward_cpu(
+    const Tensor& values,
+    TensorList offsets_list,
+    c10::IntArrayRef max_lengths,
+    const double padding_value) {
+  // TODO: Make this kernel more efficient using TensorIterator or something.
+  TORCH_INTERNAL_ASSERT(
+      offsets_list.size() == 1 && max_lengths.size() == 1,
+      "_jagged_to_padded_dense_forward(): only a single jagged dim is supported for now");
+
+  // allocate appropriately-sized padded tensor
+  auto offsets = offsets_list[0];
+  auto batch_size = offsets.size(0) - 1;
+  auto max_length = max_lengths[0];
+  auto values_shape = values.sizes().vec();
+  std::vector<int64_t> padded_shape;
+  padded_shape.reserve(values.dim() + 1);
+  padded_shape.push_back(batch_size);
+  padded_shape.push_back(max_length);
+  padded_shape.insert(padded_shape.end(), values_shape.begin() + 1, values_shape.end());
+  Tensor padded = values.new_full(padded_shape, padding_value);
+
+  // copy data to padded tensor
+  for (auto i : c10::irange(batch_size)) {
+    auto start_offset = offsets[i].item<int64_t>();
+    auto end_offset = offsets[i + 1].item<int64_t>();
```

Comment:
I think we should set `end_offset = min(offsets[i+1].item<int64_t>(), start_offset + max_length)` in case any sequence is larger than max_length. Or if we don't want to support it right now, add an assert